src/libraries/System.Text.Json/src/System/Text/Json/Serialization/JsonPropertyInfo.cs

```diff
@@ -333,34 +337,34 @@ public virtual void GetPolicies()
         public bool IsPropertyPolicy { get; protected set; }
 
         // The name from a Json value. This is cached for performance on first deserialize.
-        public byte[] JsonPropertyName { get; set; }
+        public byte[]? JsonPropertyName { get; set; }
 
         // The name of the property with any casing policy or the name specified from JsonPropertyNameAttribute.
-        public byte[] Name { get; private set; }
-        public string NameAsString { get; private set; }
+        public byte[]? Name { get; private set; }
+        public string? NameAsString { get; private set; }
 
         // Key for fast property name lookup.
         public ulong PropertyNameKey { get; set; }
 
         // Options can be referenced here since all JsonPropertyInfos originate from a JsonClassInfo that is cached on JsonSerializerOptions.
-        protected JsonSerializerOptions Options { get; set; }
+        protected JsonSerializerOptions Options { get; set; } = null!; // initialized in Init method
 
         protected abstract void OnRead(ref ReadStack state, ref Utf8JsonReader reader);
         protected abstract void OnReadEnumerable(ref ReadStack state, ref Utf8JsonReader reader);
         protected abstract void OnWrite(ref WriteStackFrame current, Utf8JsonWriter writer);
         protected virtual void OnWriteDictionary(ref WriteStackFrame current, Utf8JsonWriter writer) { }
         protected abstract void OnWriteEnumerable(ref WriteStackFrame current, Utf8JsonWriter writer);
 
-        public Type ParentClassType { get; private set; }
+        public Type? ParentClassType { get; private set; }
 
-        public PropertyInfo PropertyInfo { get; private set; }
+        public PropertyInfo? PropertyInfo { get; private set; }
```

Comment:
Should this be not null just like some of the other things that are initialized in Init method?
```suggestion
        public PropertyInfo PropertyInfo { get; private set; }
``